pycytominer/cyto_utils/cell_locations.py

```diff
@@ -0,0 +1,400 @@
+
+        image_index_str = ", ".join(self.image_index)
+
+        image_query = f"SELECT  {self.image_column},{image_index_str} FROM Image;"
+
+        nuclei_df = pd.read_sql_query(nuclei_query, conn)
+
+        image_df = pd.read_sql_query(image_query, conn)
+
+        conn.close()
+
+        # if the single_cell file was downloaded from S3, delete the temporary file
+        if "temp_single_cell_input" in locals():
+            os.remove(temp_single_cell_input)
+
+        # Merge the Image and Nuclei tables
+        merged_df = pd.merge(image_df, nuclei_df, on=self.image_column, how="inner")
+
+        # Cast the cell location columns to float
+        merged_df[self.cell_x_loc] = merged_df[self.cell_x_loc].astype(float)
+        merged_df[self.cell_y_loc] = merged_df[self.cell_y_loc].astype(float)
+
+        # Cast the object column to int
+        merged_df[self.object_column] = merged_df[self.object_column].astype(int)
+
+        # Cast the image index columns to str
+        for col in self.image_index:
+            merged_df[col] = merged_df[col].astype(str)
+
+        merged_df = self._convert_to_per_row_dict(merged_df)
+
+        return merged_df
+
+    def add_cell_location(self):
+        """Add the X,Y locations of all cells to the metadata file in the corresponding row, packed into a single column.
+        Optionally, save the augmented metadata file as a Parquet file.
+
+        Returns
+        -------
+        Pandas DataFrame
+            Either a data frame or the path to a Parquet file with the X,Y locations of all cells packed into a single column
+        """
+
+        # If self.augmented_metadata_output is not None and it is a str and the file already exists, there is nothing to do
+        if (
+            self.augmented_metadata_output is not None
+            and isinstance(self.augmented_metadata_output, str)
+            and self.overwrite is False
+            and (
+                (
+                    self.augmented_metadata_output.startswith("s3://")
+                    and self._s3_file_exists(self.augmented_metadata_output)
+                )
+                or (
+                    not self.augmented_metadata_output.startswith("s3://")
+                    and os.path.exists(self.augmented_metadata_output)
+                )
+            )
+        ):
+            return self.augmented_metadata_output
+
+        # Load the data
+        metadata_df = self._load_metadata()
+        single_cell_df = self._load_single_cell()
+
+        # Merge the data and single_cell tables
+        augmented_metadata_df = pd.merge(
+            metadata_df,
+            single_cell_df,
+            on=self.image_index,
+            how="left",
+        )
+
+        # If self.augmented_metadata_output is not None, save the data
+        if self.augmented_metadata_output is not None:
+            augmented_metadata_df.to_parquet(
```

Comment:
where possible, please use https://github.com/cytomining/pycytominer/blob/master/pycytominer/cyto_utils/output.py

Reply:
I've added a comment in here to switch to cyto_utils.output if there's a need to support more file types.